src/main/java/org/opentripplanner/netex/loader/parser/SiteFrameParser.java

```diff
@@ -109,11 +111,13 @@ private void parseTariffZones(List<JAXBElement<? extends Zone_VersionStructure>>
   private void parseQuays(Quays_RelStructure quayRefOrQuay) {
     if (quayRefOrQuay == null) return;
 
-    for (Object quayObject : quayRefOrQuay.getQuayRefOrQuay()) {
-      if (quayObject instanceof Quay) {
-        quays.add((Quay) quayObject);
-      }
-    }
+    quayRefOrQuay
+      .getQuayRefOrQuay()
+      .stream()
+      .map(JAXBElement::getValue)
+      .filter(Quay.class::isInstance)
+      .map(Quay.class::cast)
+      .forEach(quays::add);
```

Comment:
We do this in a lot of places, consider making a utility function for it. Sometimes we filter out `null` values and sometimes not, if we used the same utility function than we could include the `null` check each time.

Something like this:
```
JAXBUtils.forEachValue(quayRefOrQuay.getQuayRefOrQuay(), Quay.class, quays::add);
---
package org.opentripplanner.netex.support;
public class JAXBUtils {
  public static <S extends JAXBElement<S>,T> Stream<T> streamValue(Class<T> type, Collection<S> c) {
    return c.stream()
      .filter(Objects::nonNull)
      .map(JAXBElement::getValue)
      .filter(Objects::nonNull)
      .filter(type::isInstance)
      .map(type::cast);
  }


  public static <S extends JAXBElement<S>,T> void forEachValue(Class<T> type, Collection<S> c, Consumer<T> body) {
    streamValue(type, c).forEach(body);
  }
}
```
I added one method for returning the stream and one for the forEach case.

Reply:
Added utility method.
The null check on the wrapped value returned by JAXBElement::getValue is in practice not needed since a JAXBElement can wrap a null only if the element is nillable, and there is currently no nillable element in NeTEx (xsi:nillable is a seldom used and quite controversial XSD feature)